### PR TITLE
Further Logger changes

### PR DESCRIPTION
**Pull request by SJuliez**
Since the logger already takes the method name from the stacktrace, it can just as well take the class name and line number, simplifying calls to
`MegaMek.getLogger().error(msg);`
or
``MegaMek.getLogger().methodBegin();`

This PR also replaces all calls to the now-deprecated methods with the simplified ones.

### CODE REVIEW

**Review by Windchild292 — CHANGES_REQUESTED** [review]
This does not build

`/home/runner/work/megamek/megamek/megamek/src/megamek/common/weapons/WeaponHandler.java:1147: error: cannot find symbol
getLogger().error(getClass(), "Unexpected target type: " + target.getTargetType());`

File: megamek/src/megamek/common/logging/DefaultMmLogger.java [hunk 1]
```diff
@@ -368,36 +455,76 @@ public void warning(final Object callingObject, final String message) {
...
+     * Retrieves the name of the class calling log/info/error etc.
+     * from the stack trace.
+     *
+     * This method MUST be called from a top-level method (such as info or fatal)
+     * as it depends on the call stack depth of exactly three methods between
+     * internally calling getStackTrace and the user's code.
+     */
+    private String getCallingClass() {
+        try {
+            return Thread.currentThread().getStackTrace()[3].getClassName();
```
Comment (NickAragua) [inline megamek/src/megamek/common/logging/DefaultMmLogger.java:525]: Does dynamically retrieving this information e.g. class name, method name have any negative performance impact? Probably not, compared to disk I/O, but worth asking.

File: megamek/src/megamek/common/MechSummaryCache.java [hunk 2]
```diff
@@ -142,7 +142,7 @@ public void removeListener(Listener listener) {
     private MechSummaryCache() {
         m_nameMap = new HashMap<>();
         m_fileNameMap = new HashMap<>();
-        LOG.setLogLevel(this, LOGLVL);
+        MegaMek.getLogger().setLogLevel(LOGLVL);
```
Comment (Windchild292) [inline megamek/src/megamek/common/MechSummaryCache.java:145]: Same as before, is this setting just for this class or in general, and why is is hardcoded.

File: megamek/src/megamek/common/QuirksHandler.java [hunk 3]
```diff
@@ -228,14 +227,11 @@ public static String replaceUnitType(String unitId, String newUnitType) {
...
     private static Map<String, List<QuirkEntry>> loadQuirksFile(String path) throws IOException {
-        final String METHOD_NAME = "loadQuirksFile(String)";
-        
         Map<String, List<QuirkEntry>> quirkMap = new HashMap<>();
 
         File file = new File(path);
         if (!file.exists() || !file.isFile()) {
-            getLogger().log(QuirksHandler.class, METHOD_NAME, LogLevel.WARNING,
-                            "Could not load quirks from " + path);
+            getLogger().warning("Could not load quirks from " + path);
```
Comment (Windchild292) [inline megamek/src/megamek/common/QuirksHandler.java:234]: Keeping this using the DefaultMMLogger?

File: megamek/src/megamek/common/pathfinder/AeroGroundPathFinder.java [hunk 4]
```diff
@@ -59,29 +57,22 @@ protected int getMaximumVelocity(IAero mover) {
     protected IGame game;
     private List<MovePath> aeroGroundPaths;
     protected int maxThrust;
-    private MMLogger logger;
 
     protected AeroGroundPathFinder(IGame game) {
         this.game = game;
-        getLogger().setLogLevel(LOGGER_CATEGORY, LogLevel.DEBUG);
+        MegaMek.getLogger().setLogLevel(LogLevel.DEBUG);
```
Comment (Windchild292) [inline megamek/src/megamek/common/pathfinder/AeroGroundPathFinder.java:63]: Same as before, file or system, and why hardcoded.

**Review by Windchild292 — COMMENTED** [review]
Initial comments, currently 83/97 files done

File: megamek/src/megamek/common/logging/DefaultMmLogger.java (hunk 1 above)
Comment (Windchild292) [reply megamek/src/megamek/common/logging/DefaultMmLogger.java:525]: It shouldn't, but.. I'll test on my potato laptop before we merge the story just to be on the safe side

File: megamek/src/megamek/client/ratgenerator/RATGenerator.java [hunk 5]
```diff
@@ -884,7 +872,7 @@ private void loadFactions(File dir) {
 			DocumentBuilder db = MegaMekXmlUtil.newSafeDocumentBuilder();
 			xmlDoc = db.parse(fis);
 		} catch (Exception ex) {
-			DefaultMmLogger.getInstance().error(getClass(), METHOD_NAME, ex);
+		    MegaMek.getLogger().error(ex);
 			return;
```
Comment (Windchild292) [inline megamek/src/megamek/client/ratgenerator/RATGenerator.java:876]: Tab spacing

File: megamek/src/megamek/client/ui/swing/tileset/HexTileset.java [hunk 6]
```diff
@@ -83,7 +83,7 @@ public HexTileset(IGame g) {
         game = g;
         game.addGameListener(gameListener);
         game.getBoard().addBoardListener(this);
-        LOG.setLogLevel(this, LOGLVL);
+        MegaMek.getLogger().setLogLevel(LOGLVL);
```
[the remark this answers is not in the document]
Comment (SJuliez) [reply megamek/src/megamek/client/ui/swing/tileset/HexTileset.java:86]: Only for this class. This class used to spam lots of stuff in the log. I didnt want to remove the logs altogether, just hide them normally and have a handy way of getting them back when needed. Same in MechSummaryCache

File: megamek/src/megamek/common/pathfinder/AeroGroundPathFinder.java (hunk 4 above)
Comment (SJuliez) [reply megamek/src/megamek/common/pathfinder/AeroGroundPathFinder.java:63]: Not mine, can't say.

File: megamek/src/megamek/server/Server.java [hunk 7]
```diff
@@ -817,15 +804,15 @@ private void receivePlayerVersion(Packet packet, int connId) {
                 buf.append(System.lineSeparator()).append(System.lineSeparator());
             }
             buf.append("Client Checksum is null. Client may not have a jar file");
-            getLogger().error(getClass(), METHOD_NAME, "Client does not have a jar file");
+            MegaMek.getLogger().error("Client does not have a jar file");
```
Comment (Windchild292) [inline megamek/src/megamek/server/Server.java:807]:
```suggestion
            MegaMek.getLogger().info("Client does not have a jar file");
```

File: megamek/src/megamek/server/Server.java [hunk 8]
```diff
@@ -817,15 +804,15 @@ private void receivePlayerVersion(Packet packet, int connId) {
...
             needs = true;
         // print message indicating server doesn't have jar file
         } else if (serverChecksum == null) {
             if (!version.equals(MegaMek.VERSION)) {
                 buf.append(System.lineSeparator()).append(System.lineSeparator());
             }
             buf.append("Server Checksum is null. Server may not have a jar file");
-            getLogger().error(getClass(), METHOD_NAME, "Server does not have a jar file");
+            MegaMek.getLogger().error("Server does not have a jar file");
```
Comment (Windchild292) [inline megamek/src/megamek/server/Server.java:815]:
```suggestion
            MegaMek.getLogger().info("Server does not have a jar file");
```

File: megamek/src/megamek/server/Server.java [hunk 9]
```diff
@@ -1335,15 +1320,14 @@ public boolean loadGame(File f) {
      * @return A <code>boolean</code> value whether or not the loading was successful
      */
     public boolean loadGame(File f, boolean sendInfo) {
-        final String METHOD_NAME = "loadGame(File,boolean)";
-        getLogger().info(getClass(), METHOD_NAME, "s: loading saved game file '" + f + "'"); //$NON-NLS-1$
+        MegaMek.getLogger().info("s: loading saved game file '" + f + "'"); //$NON-NLS-1$
```
Comment (Windchild292) [inline megamek/src/megamek/server/Server.java:1323]:
```suggestion
        MegaMek.getLogger().info("s: loading saved game file '" + f + "'");
```

File: megamek/src/megamek/server/Server.java [hunk 10]
```diff
@@ -1335,15 +1320,14 @@ public boolean loadGame(File f) {
...
 
         IGame newGame;
         try (InputStream is = new FileInputStream(f); InputStream gzi = new GZIPInputStream(is)) {
             XStream xstream = SerializationHelper.getXStream();
             newGame = (IGame) xstream.fromXML(gzi);
         } catch (Exception e) {
-            getLogger().error(getClass(), METHOD_NAME, "Unable to load file: " + f, e); //$NON-NLS-1$
+            MegaMek.getLogger().error("Unable to load file: " + f, e); //$NON-NLS-1$
```
Comment (Windchild292) [inline megamek/src/megamek/server/Server.java:1330]:
```suggestion
            MegaMek.getLogger().error("Unable to load file: " + f, e);
```

File: megamek/src/megamek/server/Server.java [hunk 11]
```diff
@@ -13446,7 +13402,7 @@ private void processAttack(Entity entity, Vector<EntityAction> vector) {
         for (EntityAction ea : vector) {
             // is this the right entity?
             if (ea.getEntityId() != entity.getId()) {
-                getLogger().error(getClass(), METHOD_NAME, "attack packet has wrong attacker");
+                MegaMek.getLogger().error("attack packet has wrong attacker");
```
Comment (Windchild292) [inline megamek/src/megamek/server/Server.java:13405]:
```suggestion
                MegaMek.getLogger().error("Attack packet has wrong attacker");
```

File: megamek/src/megamek/server/Server.java [hunk 12]
```diff
@@ -15269,7 +15210,7 @@ private void removeDuplicateAttacks(int entityId) {
                     allowed--;
                 }
             } else {
-                getLogger().error(getClass(), METHOD_NAME, "removing duplicate phys attack for id#" + entityId
+                MegaMek.getLogger().error("removing duplicate phys attack for id#" + entityId
```
Comment (Windchild292) [inline megamek/src/megamek/server/Server.java:15213]:
```suggestion
                MegaMek.getLogger().error("Removing duplicate phys attack for id#" + entityId
```

File: megamek/src/megamek/server/Server.java [hunk 13]
```diff
@@ -32944,7 +32867,7 @@ private void receiveUnloadStranded(Packet packet, int connId) {
         for (int index = 0; (null != entityIds) && (index < entityIds.length); index++) {
             entity = game.getEntity(entityIds[index]);
             if (!game.getTurn().isValid(connId, entity, game)) {
-                getLogger().error(getClass(), METHOD_NAME, "server got unload stranded packet for invalid entity");
+                MegaMek.getLogger().error("server got unload stranded packet for invalid entity");
```
Comment (Windchild292) [inline megamek/src/megamek/server/Server.java:32870]:
```suggestion
                MegaMek.getLogger().error("Server got unload stranded packet for invalid entity");
```

File: megamek/src/megamek/server/Server.java [hunk 14]
```diff
@@ -33358,7 +33280,7 @@ private void checkExtremeGravityMovement(Entity entity, MoveStep step,
                                 step, moveType));
                     }
                 } else if (moveType == EntityMovementType.MOVE_JUMP) {
-                    getLogger().debug(getClass(), METHOD_NAME, "gravity move check jump: "
+                    MegaMek.getLogger().debug("gravity move check jump: "
```
Comment (Windchild292) [inline megamek/src/megamek/server/Server.java:33283]:
```suggestion
                    MegaMek.getLogger().debug("Gravity move check jump: "
```

File: megamek/src/megamek/common/logging/DefaultMmLogger.java [hunk 15]
```diff
@@ -95,32 +93,37 @@ public Logger getLogger(final String loggerName) {
...
         // Construct the message from the Throwable's message.
         String message = "";
         if (null != throwable) {
             message = throwable.getMessage();
         }
-        return log(callingClass.getName(), methodName, logLevel, message, throwable);
+        return log(className, methodName, logLevel, message, throwable);
     }
 
-    @Override
```
Comment (Windchild292) [inline megamek/src/megamek/common/logging/DefaultMmLogger.java:109]: Why remove the override?

File: megamek/src/megamek/common/logging/DefaultMmLogger.java [hunk 16]
```diff
@@ -95,32 +93,37 @@ public Logger getLogger(final String loggerName) {
...
     public <T extends Throwable> T log(final Class<?> callingClass, final String methodName,
                                        final LogLevel level, final String message, final T throwable) {
         return log(callingClass.getName(), methodName, level, message, throwable);
     }
-
-    @Override
```
Comment (Windchild292) [inline megamek/src/megamek/common/logging/DefaultMmLogger.java:115]: Why remove the override?

File: megamek/src/megamek/common/logging/DefaultMmLogger.java [hunk 17]
```diff
@@ -95,32 +93,37 @@ public Logger getLogger(final String loggerName) {
...
     public <T extends Throwable> T log(final Class<?> callingClass, final String methodName,
                                        final LogLevel logLevel, final T throwable) {
+        return log(callingClass.getName(), methodName, logLevel, throwable);
+    }
+
+    public <T extends Throwable> T log(final LogLevel logLevel, String message, final T throwable) {
+        return log(getCallingClass(), getCallingMethod(), logLevel, message, throwable);
+    }
+    
+    public <T extends Throwable> T log(final String className, final String methodName,
```
Comment (Windchild292) [inline megamek/src/megamek/common/logging/DefaultMmLogger.java:105]: Override missing

File: megamek/src/megamek/common/logging/DefaultMmLogger.java [hunk 18]
```diff
@@ -95,32 +93,37 @@ public Logger getLogger(final String loggerName) {
         return throwable;
     }
 
-    @Override
     public <T extends Throwable> T log(final Class<?> callingClass, final String methodName,
                                        final LogLevel logLevel, final T throwable) {
+        return log(callingClass.getName(), methodName, logLevel, throwable);
+    }
+
+    public <T extends Throwable> T log(final LogLevel logLevel, String message, final T throwable) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/logging/DefaultMmLogger.java:101]: Override missing

File: megamek/src/megamek/common/logging/DefaultMmLogger.java [hunk 19]
```diff
@@ -95,32 +93,37 @@ public Logger getLogger(final String loggerName) {
         return throwable;
     }
 
-    @Override
```
Comment (Windchild292) [inline megamek/src/megamek/common/logging/DefaultMmLogger.java:98]: Why remove the override?

**Review by Windchild292 — COMMENTED** [review]
The rest of my comments at 97/97 files reviewed.

**Comment by SJuliez** [comment]
> This does not build
> 
> `/home/runner/work/megamek/megamek/megamek/src/megamek/common/weapons/WeaponHandler.java:1147: error: cannot find symbol getLogger().error(getClass(), "Unexpected target type: " + target.getTargetType());`

WeaponHandler does not have this line, or any logger call.

??

**Comment by Windchild292** [comment]
I've got a strange idea... merge in latest develop.

File: megamek/src/megamek/MegaMek.java [hunk 20]
```diff
@@ -597,12 +595,12 @@ private void processUnitValidator() throws ParseException {
                 }
 
                 if (ms == null) {
-                    getLogger().error(this, new IOException(filename + " not found.  Try using \"chassis model\" for input."));
+                    getLogger().error(new IOException(filename + " not found.  Try using \"chassis model\" for input."));
```
[the remark this answers is not in the document]
Comment (SJuliez) [reply megamek/src/megamek/MegaMek.java:598]: I noticed it but seeing the number of changes for this one, I simply didn't want to think about it. Another time :)

File: megamek/src/megamek/common/QuirksHandler.java (hunk 3 above)
Comment (SJuliez) [reply megamek/src/megamek/common/QuirksHandler.java:234]: This has a setlogger method. I didn't want to just remove it.

```
// Use to pass in a fake logger for unit tests.
    static void setLogger(final MMLogger newLogger) {
        logger = newLogger;
    }
```

File: megamek/src/megamek/common/logging/DefaultMmLogger.java (hunk 15 above)
Comment (SJuliez) [reply megamek/src/megamek/common/logging/DefaultMmLogger.java:109]: Readded all the Overrides

**Comment by SJuliez** [comment]
OK, added in the missing Overrides and suggestions.

File: megamek/src/megamek/client/ui/swing/HelpDialog.java [hunk 21]
```diff
@@ -1,5 +1,6 @@
 /*  
-* MegaMek - Copyright (C) 2020 - The MegaMek Team  
+* MegaMek - Copyright (C) 2013-2020 - The MegaMek Team  
+
```
Comment (Windchild292) [inline megamek/src/megamek/client/ui/swing/HelpDialog.java:3]:
```suggestion
```

**Review by Windchild292 — APPROVED** [review]
Changes look great

**Pull request merged**